ParProcCo/utils.py

```diff
@@ -22,6 +22,14 @@ def check_jobscript_is_readable(jobscript: Path) -> Path:
         raise
     return jobscript
 
+def get_filepath_on_path(filename: str) -> Path:
+    path = os.environ['PATH']
+    paths = path.split(':')
+    for p in paths:
+        for root, dir, files in os.walk(p):
```

Comment:
Should not walk directory, just test if file exists in directory. Also look at using a generator expression with `next`.